packages/hash/api/src/db/postgres/util.test.ts

```diff
@@ -0,0 +1,66 @@
+import { transaction } from "./util";
+import { ConnectionKind, PoolConnection, TransactionConnection } from "./types";
+
+describe("transaction flattening", () => {
```

Comment:
Ideally we could also check that the SQL itself doesn't contain any savepoints, but this already is quite coupled to slonik.